vaadin-date-time-picker-flow-demo/src/main/java/com/vaadin/flow/component/datetimepicker/demo/DateTimePickerView.java

```diff
@@ -0,0 +1,114 @@
+        addCard("Additional code used in the demo",
+                new Label("These methods are used in the demo."));
+    }
+
+    private void createSimpleDateTimePicker() {
+        Div message = createMessageDiv("simple-picker-message");
+
+        // begin-source-example
+        // source-example-heading: Simple date time picker
+        DateTimePicker dateTimePicker = new DateTimePicker();
+
+        dateTimePicker.addValueChangeListener(
+                event -> updateMessage(message, dateTimePicker));
+        // end-source-example
+
+        dateTimePicker.setId("simple-picker");
+        addCard("Simple date time picker", dateTimePicker, message);
+    }
+
+    private void createLocaleChangeDateTimePicker() {
+        Div message = createMessageDiv("Customize-locale-picker-message");
+        // begin-source-example
+        // source-example-heading: Date picker with customize locales
+        // By default, the dateTimePicker uses the current UI locale
+        DateTimePicker dateTimePicker = new DateTimePicker();
+        NativeButton locale1 = new NativeButton("Locale: US");
+        NativeButton locale2 = new NativeButton("Locale: UK");
+        NativeButton locale3 = new NativeButton("Locale: CHINA");
+
+        locale1.addClickListener(e -> {
+            dateTimePicker.setLocale(Locale.US);
+            updateMessage(message, dateTimePicker);
+        });
+        locale2.addClickListener(e -> {
+            dateTimePicker.setLocale(Locale.UK);
+            updateMessage(message, dateTimePicker);
+        });
+        locale3.addClickListener(e -> {
+            dateTimePicker.setLocale(Locale.CHINA);
+            updateMessage(message, dateTimePicker);
+        });
+
+        dateTimePicker.addValueChangeListener(
+                event -> updateMessage(message, dateTimePicker));
+        // end-source-example
+        locale1.setId("Locale-US");
+        locale2.setId("Locale-UK");
+        dateTimePicker.setId("locale-change-picker");
+        addCard("Date picker with customize locales", dateTimePicker, locale1,
+                locale2, locale3, message);
+    }
+
+    // begin-source-example
+    // source-example-heading: Additional code used in the demo
+    private void updateMessage(Div message, DateTimePicker dateTimePicker) {
+        LocalDateTime selectedDateTime = dateTimePicker.getValue();
+        if (selectedDateTime != null) {
+            final DateTimeFormatter formatter = DateTimeFormatter
+                    .ofPattern("yyyy-MM-dd HH:mm:ss");
+            message.setText("Value: " + selectedDateTime.format(formatter)
+                    + "\n" + "Locale: " + dateTimePicker.getLocale());
+        } else {
+            message.setText("No date is selected");
+        }
+    }
+
+    private Div createMessageDiv(String id) {
+        Div message = new Div();
+        message.setId(id);
+        message.getStyle().set("whiteSpace", "pre");
+        return message;
+    }
```

Comment:
This method is not used inside the source examples and I think it should be showed to the users either.

Reply:
Good point. I didn't notice it wasn't directly used in the demo snippets. Same issue in DatePicker demos. I moved it out of the "source-example" block now.